File: backend/app/schemas/assets.py
from datetime import datetime

from pydantic import BaseModel


class AssetRecord(BaseModel):
    id: str
    name: str
    source_kind: str
    module_kind: str | None = None
    visibility: str
    owner_user_id: str | None = None
    owner_username: str | None = None
    storage_url: str
    preview_url: str | None = None
    mime_type: str | None = None
    file_size: int | None = None
    metadata: dict[str, object] | None = None
    can_delete: bool = False
    can_publish: bool = False
    can_unpublish: bool = False
    created_at: datetime


class AssetListResponse(BaseModel):
    items: list[AssetRecord]

File: frontend/src/types/assets.ts
export interface PersistedAssetItem {
  id: string;
  name: string;
  source_kind: string;
  module_kind: string | null;
  visibility: "private" | "community";
  owner_user_id: string | null;
  owner_username: string | null;
  storage_url: string;
  preview_url: string | null;
  mime_type: string | null;
  file_size: number | null;
  metadata?: Record<string, unknown> | null;
  can_delete: boolean;
  can_publish: boolean;
  can_unpublish: boolean;
  created_at: string;
}

export interface PersistedAssetResponse {
  items: PersistedAssetItem[];
}
